### ISSUE TITLE
MAP types are not supported on version 21.0

### ISSUE THREAD

**Issue by magorlick** [rilldata/rill-developer#1785]
**Describe the bug**
We recently released support for nested types during profiling, however I am finding that MAPs do not have profiling insights or table outputs.

**To Reproduce**
1. start Rill
2. model `map([100, 5], [42, 43])`
3. see no table, no profiling

**Expected behavior**
I expect to see top k and tables for MAP types.

**Screenshots**
Internal video here: https://www.notion.so/rilldata/MAP-types-not-working-in-21-0-62a1e1638d524411ae7483e675328bb0

**Comment by hamilton** [comment]
This appears to possibly an invalid type in the runtime. I get a 400 error with `"proto: invalid type: duckdb.Map"`. Passing on to @begelundmuller to triage, if my understanding is correct.

**Comment by begelundmuller** [comment]
Yep, looks like we didn't have proto/JSON serialization for DuckDB map types. Added it here: https://github.com/rilldata/rill-developer/pull/1787

**Comment by magorlick** [comment]
This is not working for me in the application @begelundmuller .  Please see my note in notion. https://www.notion.so/rilldata/Application-Bugs-and-Polish-for-21-1-6d7f59f3076249f7af64dfbc8acc47df

**Comment by begelundmuller** [comment]
It should work for everything except maps with map/array *keys* (not values). Conversation for supporting complex types in map keys continuing here: https://github.com/rilldata/rill-developer/issues/1798.
